tests/unit/db/view_execution_tests.py

```diff
@@ -0,0 +1,628 @@
+#!/usr/bin/env python
+# Copyright (c) 2016 IBM. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the execution of view queries using translated parameters.
+"""
+import unittest
+
+from cloudant.design_document import DesignDocument
+from .unit_t_db_base import UnitTestDbBase
+
+class QueryParmExecutionTests(UnitTestDbBase):
+    """
+    Test cases for the execution of views queries using translated parameters.
+    """
+
+    def setUp(self):
+        """
+        Set up test attributes
+        """
+        super(QueryParmExecutionTests, self).setUp()
+        self.db_set_up()
+        self.ddoc = DesignDocument(self.db, 'ddoc001')
+        self.ddoc.add_view(
+            'view001',
+            'function (doc) {\n emit(doc._id, 1);\n}'
+        )
+        self.ddoc.add_view(
+            'view002',
+            'function (doc) {\n emit(doc._id, 1);\n}',
+            '_count'
+        )
+        self.ddoc.add_view(
+            'view003',
+            'function (doc) {\n emit(Math.floor(doc.age / 2), 1);\n}'
+        )
+        self.ddoc.add_view(
+            'view004',
+            'function (doc) {\n emit(Math.floor(doc.age / 2), 1);\n}',
+            '_count'
+        )
+        self.ddoc.add_view(
+            'view005',
+            'function (doc) {\n emit([doc.name, doc.age], 1);\n}'
+        )
+        self.ddoc.save()
+        self.view001 = self.ddoc.get_view('view001')
+        self.view002 = self.ddoc.get_view('view002')
+        self.view003 = self.ddoc.get_view('view003')
+        self.view004 = self.ddoc.get_view('view004')
+        self.view005 = self.ddoc.get_view('view005')
+
+    def tearDown(self):
+        """
+        Reset test attributes
+        """
+        self.db_tear_down()
+        super(QueryParmExecutionTests, self).tearDown()
+
+    def test_descending_true(self):
+        """
+        Test view query using descending parameter set to True.
+
+        The view used here will generate rows of data where each key will equal
+        the document id.  Such as:
+        {'key': 'julia000', 'id': 'julia000', 'value': 1},
+        {'key': 'julia001', 'id': 'julia001', 'value': 1},
+        {'key': 'julia002', 'id': 'julia002', 'value': 1},
+        ...
+        """
+        self.populate_db_with_documents(100)
```

Comment:
Since `setUp` is run before each test case, why not move `self.populate_db_with_documents` into it?
```
When a setUp() method is defined, the test runner will run that method prior to each test.
```
Supporting link: https://docs.python.org/2.6/library/unittest.html